File: api/cadangan-api/src/main/java/my/emasjid/cadanganapi/entity/Cadangan.java
package my.emasjid.cadanganapi.entity;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.ManyToOne;
import jakarta.persistence.Table;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@NoArgsConstructor
@Entity
@Table(name = "cadangan")
public class Cadangan {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @ManyToOne
    @JoinColumn(name = "cadangan_types_id", referencedColumnName = "id")
    private CadanganType cadanganType;

    @Column(name = "cadangan_text", nullable = false, length = 1024)
    private String cadanganText;

    @Column(name = "tindakan_text", length = 1024)
    private String tindakanText;

    @Column(name = "is_open", nullable = false)
    private Boolean isOpen = false;

    @Column(name = "score")
    private Short score;

    @Column(name = "create_date")
    private Long createDate;
}

File: api/cadangan-public-api-go/model/cadangan.go
package model

type Cadangan struct {
	ID            int          `db:"id" json:"id"`
	CadanganType  CadanganType `gorm:"embedded" json:"cadanganType"`
	CadanganText  string       `db:"cadangan_text" json:"cadanganText"`
	TindakanText  string       `db:"tindakan_text" json:"tindakanText"`
	CadanganNama  string       `db:"cadangan_nama" json:"cadanganNama"`
	CadanganEmail string       `db:"cadangan_email" json:"cadanganEmail"`
	CadanganPhone string       `db:"cadangan_phone" json:"cadanganPhone"`
	IsOpen        bool         `db:"is_open" json:"isOpen"`
	Score         int          `db:"score" json:"score"`
	CreateDate    int64        `db:"create_date" json:"createDate"`
}
